contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -1402,12 +1407,29 @@ translate_char_pos(const char* str,		/* UTF-8 string */
 }
 
 static int
-icu_compare_utf8_coll(pg_locale_t locale, UChar *uchar1, int32_t ulen1,
-					  UChar *uchar2, int32_t ulen2)
+icu_compare_utf8_coll(UCollator  *coll, UChar *uchar1, int32_t ulen1,
+					  UChar *uchar2, int32_t ulen2, bool is_cs_ai)
 {
-	return ucol_strcoll(locale->info.icu.ucol,
-							uchar1, ulen1,
-							uchar2, ulen2);
+	UErrorCode	status = U_ZERO_ERROR;
+	UCollator   *cloned_coll = ucol_safeClone(coll, NULL, NULL, &status);
+
+	if (U_FAILURE(status))
+		ereport(ERROR,
+			(errcode(ERRCODE_INTERNAL_ERROR),
+				errmsg("failed to clone collator in icu compare: %s", u_errorName(status))));
+
+	if (is_cs_ai)
+	{
+		ucol_setAttribute(cloned_coll, UCOL_CASE_FIRST, UCOL_OFF, &status);
+		if (U_FAILURE(status))
+			ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+					errmsg("failed to set attribute for ucollator: %s", u_errorName(status))));
+	}
```

Comment:
We do this for range check for CS_AI collations. We defined the UPPER case as first in sort order while creating CS_AI collations which is why the below returns a match 
`SELECT PATINDEX('[A-Z]', 'b' COLLATE Latin1_General_CI_AI)`
even when it shouldn't.

So we disbale the CASE_FIRST attribute for range check wildcards

Reply:
Nice!